diff --git a/dist/src/lib/commands/run/createRun.js b/dist/src/lib/commands/run/createRun.js
index cd24550487e93d23848499a51fc4350d460ba799..c620ba8d116987b8475baf628550d940a9533bf7 100644
--- a/dist/src/lib/commands/run/createRun.js
+++ b/dist/src/lib/commands/run/createRun.js
@@ -148,6 +148,7 @@ ${devices
     }
     else {
         const bootedDevices = devices.filter(({ state, type }) => state === 'Booted' && type === deviceOrSimulator);
+        console.log('Available devices', devices);
         if (bootedDevices.length === 0) {
             // fallback to present all devices when no device is selected
             if (isInteractive()) {
