diff --git a/dist/src/lib/commands/run/runOnSimulator.js b/dist/src/lib/commands/run/runOnSimulator.js
index c4012b1ca5876e6830606047063b727c98bd7c91..e86de1d43f86951d9b1a8d1a01e5a32edfd94721 100644
--- a/dist/src/lib/commands/run/runOnSimulator.js
+++ b/dist/src/lib/commands/run/runOnSimulator.js
@@ -44,6 +44,7 @@ async function bootSimulator(selectedSimulator) {
         await spawn('xcrun', ['simctl', 'boot', selectedSimulator.udid]);
     }
     catch (error) {
+        console.log(error);
         if (
         // It may happen on GitHub Actions when the simulator is already booted,
         // even though the simctl returns its state as Shutdown
@@ -51,9 +52,9 @@ async function bootSimulator(selectedSimulator) {
             logger.debug(`Simulator ${selectedSimulator.udid} already booted. Skipping.`);
             return;
         }
-        throw new RnefError('Failed to boot Simulator', {
-            cause: error.stderr,
-        });
+        // throw new RnefError('Failed to boot Simulator', {
+        //     cause: error.stderr,
+        // });
     }
 }
 export default async function installAppOnSimulator(udid, binaryPath) {
