diff --git a/dist/src/lib/commands/run/runOnSimulator.js b/dist/src/lib/commands/run/runOnSimulator.js
index c4012b1ca5876e6830606047063b727c98bd7c91..b185fa83f771344b44ec5da622c4c636fe13e67f 100644
--- a/dist/src/lib/commands/run/runOnSimulator.js
+++ b/dist/src/lib/commands/run/runOnSimulator.js
@@ -51,9 +51,10 @@ async function bootSimulator(selectedSimulator) {
             logger.debug(`Simulator ${selectedSimulator.udid} already booted. Skipping.`);
             return;
         }
-        throw new RnefError('Failed to boot Simulator', {
-            cause: error.stderr,
-        });
+        // error.stderr is always empty when running on GitHub Actions
+        // throw new RnefError('Failed to boot Simulator', {
+        //     cause: error.stderr,
+        // });
     }
 }
 export default async function installAppOnSimulator(udid, binaryPath) {
