diff --git a/Libraries/Utilities/HMRClient.js b/Libraries/Utilities/HMRClient.js
index 776056cf527553fb659c0de31ab376342deddb10..994d1b100fdc230eb954e61a2f08c84cc8a29382 100644
--- a/Libraries/Utilities/HMRClient.js
+++ b/Libraries/Utilities/HMRClient.js
@@ -101,6 +101,9 @@ const HMRClient: HMRClientNativeInterface = {
 
   registerBundle(requestUrl: string) {
     invariant(hmrClient, 'Expected HMRClient.setup() call at startup.');
+    const host = hmrClient._ws.url.split('/hot')[0];
+    // skip registerBundle calls from other hosts
+    if (!requestUrl.startsWith(host)) return;
     pendingEntryPoints.push(requestUrl);
     registerBundleEntryPoints(hmrClient);
   },
